server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1792,6 +1792,49 @@ public long recoverLocallyUpToGlobalCheckpoint() {
         }
     }
 
+    public long recoverLocallyAndFetchStartSeqNo(boolean localTranslog) {
+        if (localTranslog) {
+            return recoverLocallyUpToGlobalCheckpoint();
+        } else {
+            return recoverLocallyUptoLastCommit();
+        }
+    }
+
+    /**
+     * The method figures out the sequence number basis the last commit.
+     *
+     * @return the starting sequence number from which the recovery should start.
+     */
+    public long recoverLocallyUptoLastCommit() {
+        long seqNo;
+        assert Thread.holdsLock(mutex) == false : "recover locally under mutex";
+        if (state != IndexShardState.RECOVERING) {
+            throw new IndexShardNotRecoveringException(shardId, state);
+        }
+        recoveryState.validateCurrentStage(RecoveryState.Stage.INDEX);
+        assert routingEntry().recoverySource().getType() == RecoverySource.Type.PEER : "not a peer recovery [" + routingEntry() + "]";
+
+        try {
+            seqNo = Long.parseLong(store.readLastCommittedSegmentsInfo().getUserData().get(SequenceNumbers.MAX_SEQ_NO));
+        } catch (org.apache.lucene.index.IndexNotFoundException e) {
```

Comment:
nit: will prefer import. Also add an assertion that this will be called on remote translog enabled indices

Reply:
Did you mean to use static import for MAX_SEQ_NO? If so, made that change. Added the assertions.